typings/resin-sdk.d.ts

```diff
@@ -0,0 +1,590 @@
+		status: BuildStatus;
+		update_timestamp: string | null;
+
+		belongs_to__user: NavigationResource<User>;
+		belongs_to__application: NavigationResource<Application>;
+	}
+
+	interface BillingAccountAddressInfo {
+		address1: string;
+		address2: string;
+		city: string;
+		state: string;
+		zip: string;
+		country: string;
+		phone: string;
+	}
+
+	interface BillingAccountInfo {
+		account_state: string;
+		first_name: string;
+		last_name: string;
+		company_name: string;
+		cc_emails: string;
+		vat_number: string;
+		address: BillingAccountAddressInfo;
+	}
+
+	type BillingInfoType = 'bank_account' | 'credit_card' | 'paypal';
+
+	interface BillingInfo {
+		full_name: string;
+
+		first_name: string;
+		last_name: string;
+		company: string;
+		vat_number: string;
+		address1: string;
+		address2: string;
+		city: string;
+		state: string;
+		zip: string;
+		country: string;
+		phone: string;
+
+		type?: BillingInfoType;
+	}
+
+	interface CardBillingInfo extends BillingInfo {
+		card_type: string;
+		year: string;
+		month: string;
+		first_one: string;
+		last_four: string;
+	}
+
+	interface BankAccountBillingInfo extends BillingInfo {
+		account_type: string;
+		last_four: string;
+		name_on_account: string;
+		routing_number: string;
+	}
+
+	interface TokenBillingSubmitInfo {
+		token_id: string;
+	}
+
+	interface BillingPlanInfo {
+		name: string;
+		billing?: BillingPlanBillingInfo;
+	}
+
+	interface BillingPlanBillingInfo {
+		currency: string;
+		currencySymbol?: string;
+	}
+
+	interface InvoiceInfo {
+		closed_at: string;
+		created_at: string;
+		currency: string;
+		invoice_number: string;
+		subtotal_in_cents: string;
+		total_in_cents: string;
+		uuid: string;
+	}
+
+	interface Device {
+		is_on__commit: string;
+		created_at: string;
+		device_type: string;
+		id: number;
+		name: string;
+		os_version: string;
+		os_variant?: string;
+		status_sort_index?: number;
+		uuid: string;
+		ip_address: string | null;
+		vpn_address: string | null;
+		last_connectivity_event: string;
+		is_in_local_mode: boolean;
+		app_name: string;
+		state: { key: string; name: string };
+		status: string;
+		provisioning_state: string;
+		is_online: boolean;
+		is_connected_to_vpn: boolean;
+		is_locked_until__date: string;
+		supervisor_version: string;
+		should_be_managed_by__supervisor_release: number;
+		is_web_accessible: boolean;
+		has_dependent: boolean;
+		note: string;
+		location: string;
+		latitude?: string;
+		longitude?: string;
+		custom_latitude?: string;
+		custom_longitude?: string;
+		is_accessible_by_support_until__date: string;
+		download_progress?: number;
+		provisioning_progress?: number;
+
+		belongs_to__application: NavigationResource<Application>;
+		belongs_to__user: NavigationResource<User>;
+		is_managed_by__device: NavigationResource<Device>;
+
+		device_environment_variable: ReverseNavigationResource<
+			DeviceEnvironmentVariable
+		>;
+		manages__device: ReverseNavigationResource<Device>;
+		local_id?: string;
```

Comment:
Can we move this below `provisioning_progress` so that these props are grouped by primitives/navigation/reverseNavigation? (have that change in my tags PR)

Reply:
Done